care/facility/api/serializers/shifting.py

```diff
@@ -182,16 +244,25 @@ def create(self, validated_data):
 
         validated_data["is_kasp"] = False
 
-        orgin_facility_external_id = validated_data.pop("orgin_facility")["external_id"]
+        # orgin_facility_external_id = validated_data.pop("orgin_facility")["external_id"]
         # validated_data["orgin_facility_id"] = Facility.objects.get(external_id=orgin_facility_external_id).id
 
-        shifting_approving_facility_external_id = validated_data.pop("shifting_approving_facility")["external_id"]
+        shifting_approving_facility_external_id = validated_data.pop(
+            "shifting_approving_facility"
+        )["external_id"]
         validated_data["shifting_approving_facility_id"] = Facility.objects.get(
             external_id=shifting_approving_facility_external_id
         ).id
 
-        if "assigned_facility" in validated_data:
-            assigned_facility_external_id = validated_data.pop("assigned_facility")["external_id"]
+        if validated_data.get("assigned_facility_external"):
+            validated_data["assigned_facility"] = None
+        elif validated_data.get("assigned_facility"):
+            validated_data["assigned_facility_external"] = None
```

Comment:
If both dont exist then we need a validation error right?

Reply:
This is already being done if both don't exist.
https://github.com/coronasafe/care/pull/1246/files#diff-e73c078ebb0722d6c6cbbdf3071b60a55aaa9945316336701e6b4e9c934fcbc8R209

```
       if "status" in validated_data:
            if validated_data["status"] in RECIEVING_REQUIRED_STATUS:
                if (
                    not (
                        instance.assigned_facility
                        or instance.assigned_facility_external
                    )
                ) and (not assigned):
                    raise ValidationError(
                        {
                            "status": [
                                "Destination Facility is required for moving to this stage."
                            ]
                        }
                    )
```